connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClientCache.scala

```diff
@@ -10,70 +10,67 @@ import com.microsoft.kusto.spark.utils.{KustoConstants => KCONST}
         )
       case keyVaultParams: KeyVaultAuthentication =>
         val app = KeyVaultUtils.getAadAppParametersFromKeyVault(keyVaultParams)
         (
-          ConnectionStringBuilder.createWithAadApplicationCredentials(aliasAndAuth.engineUri, app.ID, app.password, app.authority),
-          ConnectionStringBuilder.createWithAadApplicationCredentials(aliasAndAuth.ingestUri, app.ID, app.password, app.authority)
+          ConnectionStringBuilder.createWithAadApplicationCredentials(clusterAndAuth.engineUri, app.ID, app.password, app.authority),
+          ConnectionStringBuilder.createWithAadApplicationCredentials(clusterAndAuth.ingestUri, app.ID, app.password, app.authority)
         )
       case userPrompt: KustoUserPromptAuthentication => (
         // TODO authoirty
-        ConnectionStringBuilder.createWithUserPrompt(aliasAndAuth.engineUri, userPrompt.authority),
-        ConnectionStringBuilder.createWithUserPrompt(aliasAndAuth.ingestUri, userPrompt.authority)
+        ConnectionStringBuilder.createWithUserPrompt(clusterAndAuth.engineUri, userPrompt.authority, null),
+        ConnectionStringBuilder.createWithUserPrompt(clusterAndAuth.ingestUri, userPrompt.authority, null)
       )
       case userToken: KustoAccessTokenAuthentication => (
-        ConnectionStringBuilder.createWithAadAccessTokenAuthentication(aliasAndAuth.engineUri, userToken.token),
-        ConnectionStringBuilder.createWithAadAccessTokenAuthentication(aliasAndAuth.ingestUri, userToken.token)
+        ConnectionStringBuilder.createWithAadAccessTokenAuthentication(clusterAndAuth.engineUri, userToken.token),
+        ConnectionStringBuilder.createWithAadAccessTokenAuthentication(clusterAndAuth.ingestUri, userToken.token)
       )
       case tokenProvider: KustoTokenProviderAuthentication => (
-        ConnectionStringBuilder.createWithAadTokenProviderAuthentication(aliasAndAuth.engineUri, tokenProvider.tokenProviderCallback),
-        ConnectionStringBuilder.createWithAadTokenProviderAuthentication(aliasAndAuth.ingestUri, tokenProvider.tokenProviderCallback)
+        ConnectionStringBuilder.createWithAadTokenProviderAuthentication(clusterAndAuth.engineUri, tokenProvider.tokenProviderCallback),
+        ConnectionStringBuilder.createWithAadTokenProviderAuthentication(clusterAndAuth.ingestUri, tokenProvider.tokenProviderCallback)
       )
     }
 
     engineKcsb.setClientVersionForTracing(KCONST.ClientName)
     ingestKcsb.setClientVersionForTracing(KCONST.ClientName)
 
-    new KustoClient(aliasAndAuth.clusterAlias, engineKcsb, ingestKcsb)
+    new KustoClient(engineKcsb, ingestKcsb, clusterAndAuth.clusterAlias)
   }
 
-  private[kusto] case class AliasAndAuth(clusterAlias: String, engineUrl: String, authentication: KustoAuthentication) {
+  private[kusto] case class ClusterAndAuth(engineUrl: String, authentication: KustoAuthentication, ingestionUri: Option[String], clusterAlias:String) {
```

Comment:
I see you changed the order of parameters for some methods in this PR. Why not here as well?

I'd think it should be something like:
clusterAlias:String, authentication: KustoAuthentication, engineUrl: String, ingestionUri: Option[String]

Reply:
This is the main class that caused this change - i wanted alias to be last as it is less important

Reply:
so its actually did change accordingly

Reply:
`alias` can be last, but I think it'd be best to improve the order. Specifically, `engineUrl: String` should be adjacent to `ingestionUri: Option[String]`